llvm/lib/Transforms/IPO/SampleProfile.cpp

```diff
@@ -2411,10 +2411,10 @@ void SampleProfileMatcher::runOnFunction(const Function &F) {
   if (ReportProfileStaleness || PersistProfileStaleness)
     recordCallsiteMatchStates(F, IRAnchors, ProfileAnchors, nullptr);
 
-  // Run profile matching for checksum mismatched profile, currently only
-  // support for pseudo-probe.
-  if (SalvageStaleProfile && FunctionSamples::ProfileIsProbeBased &&
-      !ProbeManager->profileIsValid(F, *FSFlattened)) {
+  // Run profile matching for checksum mismatched profile. For probe-based
```

Comment:
Nit: Rephrase or remove this "Run profile matching for checksum mismatched profile", since non-probe doesn't have checksum.

Reply:
Done.